bson/bson_test.go

```diff
@@ -179,6 +204,25 @@ func TestMapCodec(t *testing.T) {
 		assert.Equal(t, mapObj, got, "expected result %v, got %v", mapObj, got)
 
 	})
+
+	t.Run("keys implements encoding.TextMarshaler and encoding.TextUnmarshaler", func(t *testing.T) {
+		mapObj := map[keyStruct]int{
+			keyStruct{val: 10}: 100,
```

Comment:
```suggestion
			{val: 10}: 100,
```
Remove redundant `keyStruct` type.